integration/tests/lib/block_state.go

```diff
@@ -136,7 +136,9 @@ func (bs *BlockState) processAncestors(t *testing.T, b *messages.BlockProposal,
 // latest finalized block is 10, this will wait until any block height higher than 10 is finalized, and
 // returns that block.
 func (bs *BlockState) WaitForHighestFinalizedProgress(t *testing.T) *messages.BlockProposal {
```

Comment:
This function seems like a flaky test generator.  In particular, currentFinalized could be have been updated even before WaitForHighestFinalizedProgress got executed.  We should pass in the currentFinalized explicitly.

Reply:
Yeah, it seemed quite bad for me at first glance usages are quite legit: it is used as a healthcheck for the network and bootstraps the `currentFinalized` block that is used thereafter in tests.

Reply:
Renamed to `WaitForInitialization` to highlight its intended usage.